installer/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/workflow/UpdateWorkflow.java

```diff
@@ -390,16 +390,29 @@ protected Collection<UpdateAction> getUpdateActionsFromPaths(List<Path> changedP
                     }
                     // now add targets if necessary
                     switch (pathAction) {
+                        case RESOURCES: {
+                            String target = changedPath.getName(i + 1).toString();
```

Comment:
Could throw IllegalArgumentException

We could grab `target` before the switch statement?